overlord/assertstate/validation_set_tracking.go

```diff
@@ -121,10 +122,19 @@ func verifyForgetAllowedByModelAssertion(st *state.State, accountID, name string
 	return nil
 }
 
+// ForgetValidationSetOpts holds options for ForgetValidationSet.
+type ForgetValidationSetOpts struct {
+	// ForceForget is used to forget a validation set even if it's enforced by
+	// the model. This is currently used in the case that we are rolling back a
+	// remodel and we need to replace a validation set that is enforced by the
+	// new model.
+	ForceForget bool
+}
+
 // ForgetValidationSet deletes a validation set for the given accountID and name.
 // It is not an error to delete a non-existing one. If the validation-set
 // is controlled by the model assertion it may not be allowed to forget it.
-func ForgetValidationSet(st *state.State, accountID, name string) error {
+func ForgetValidationSet(st *state.State, accountID, name string, opts ForgetValidationSetOpts, deviceCtx snapstate.DeviceContext) error {
```

Comment:
These changes will need unit tests, if it is the route we want to go. I also wonder if we need a function to remove and add validation sets at the same time, so there isn't a gap of having no validation set.

Reply:
Okay, in this case we hold the lock so it won't matter.